### PR TITLE
bug/CSI-2431_delete_source_PVC_failed

### PR DESCRIPTION
**Pull request by ArbelNathan**

### CODE REVIEW

**Review by lisyukanna — APPROVED** [review]
LGTM

File: controller/tests/array_action/svc/array_mediator_svc_test.py [hunk 1]
```diff
@@ -216,8 +216,7 @@ def test_delete_volume_still_copy_fcmaps_not_removed(self):
             self.svc.delete_volume("vol")
 
     def test_delete_volume_has_clone_fcmaps_removed(self):
-        fcmaps_as_target = self.fcmaps
-        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=fcmaps_as_target), Mock(as_list=[])]
+        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=[]), Mock(as_list=self.fcmaps_as_source)]
```
Comment (oriyarde) [inline controller/tests/array_action/svc/array_mediator_svc_test.py:219]: please add a meaningful name for each mock here. e.g. fcmaps_before_x

**Review by erantzabari — APPROVED** [review]

File: controller/tests/array_action/svc/array_mediator_svc_test.py (hunk 1 above)
Comment (ArbelNathan) [reply controller/tests/array_action/svc/array_mediator_svc_test.py]: done

File: controller/tests/array_action/svc/array_mediator_svc_test.py [hunk 2]
```diff
@@ -201,23 +201,28 @@ def _prepare_mocks_for_object_still_in_use(self):
...
         fcmaps_as_source[0].copy_rate = "0"
-        self.svc.client.svcinfo.lsfcmap.side_effect = [Mock(as_list=[]), Mock(as_list=fcmaps_as_source)]
+        fcmaps_list_as_source = Mock(as_list=fcmaps_as_source)
+        self.svc.client.svcinfo.lsfcmap.side_effect = [fcmaps_list_as_target, fcmaps_list_as_source]
         with self.assertRaises(array_errors.ObjectIsStillInUseError):
             self.svc.delete_volume("vol")
 
     def test_delete_volume_still_copy_fcmaps_not_removed(self):
         self._prepare_mocks_for_object_still_in_use()
+        fcmaps_list_as_target = Mock(as_list=[])
```
Comment (oriyarde) [inline controller/tests/array_action/svc/array_mediator_svc_test.py:214]: I would avoid using the type in the variable name

Comment (ArbelNathan) [reply controller/tests/array_action/svc/array_mediator_svc_test.py:214]: its not about the type, just making sure that we are clear about what its for

Comment (oriyarde) [reply controller/tests/array_action/svc/array_mediator_svc_test.py:214]: it can be both clear and without the unnecessary type

Comment (ArbelNathan) [reply controller/tests/array_action/svc/array_mediator_svc_test.py]: done

**Review by oriyarde — APPROVED** [review]
LGTM

**Review by erantzabari — APPROVED** [review]

**Review by lisyukanna — APPROVED** [review]

**Pull request merged**